docs/tech_specs/monitoring_data_consumption.md

```diff
@@ -0,0 +1,61 @@
+
+To give a rough idea of the required resources for processing the message logs, the CSV files may be up to ~200MB which is based on the number of messages we send to ARINC over the course of a day. While this ultimately isn’t a ton of data, we should note that this is only counting visual messages. We also send audio messages, but they are not able to be logged yet. If/when those are added, the number of rows in the CSV would at least double.
+
+As a list of steps, the solution would involve the following in sequence: 
+
+  1. ARINC builds out an application that filters out unneeded logs and compiles the data that we outlined in our specification. With CTD having improved the usability of the `uid` field that we send in our messages to the ARINC headend server, this is achievable now.
+
+  2. ARINC sets up an endpoint for CTD to hit once per day around 2AM to receive a file containing the compiled logs. It would probably be good if the file is a zip to keep the size of data being transferred between servers small.
+
+  3. CTD sets up a recurring job which calls ARINC’s endpoint nightly to receive the file and dump it into an S3 bucket. Having the logs stored in S3 also makes it convenient if we ever need to dig up past log files for some reason.
+
+  4. CTD sets up an application that is responsible for pulling down the zip file from S3, unzipping it, and then streaming it into Splunk. There may be different options to explore on how to achieve this step. Splunk does not provide an easy way of uploading a CSV file through their API. They do provide a mechanism called a Universal Forwarder meant for forwarding large amounts of data into Splunk, though past attempts at setting these up have not been successful at the CTD. A "brute force" option could be to stream the CSV and `Logger.log()` the rows into Splunk, but the amount of requests would be close to 600,000 for one CSV for visual messages only.
+
+The main reason for standing up a new application would be to avoid resource consumption on opstech3 and it's an effort to edge PA/ESS features towards AWS rather than increasing our dependence on on-prem servers.
+
+An alternative, potentially simpler option _could_ be to try streaming the CSV into Splunk directly from Realtime Signs and skip standing up an additional application since that will require some additional time and resources. We could still dump the zips or raw CSVs in S3 if we want to keep a record of what we receive from ARINC.
```

Comment:
The other downside to this is that it might make the logs for Realtime Signs harder to work with, unless the logs would also be streaming from the PA/ESS (which it doesn't sound like).